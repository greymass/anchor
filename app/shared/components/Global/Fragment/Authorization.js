// @flow
import React, { Component } from 'react';
import { Popup } from 'semantic-ui-react';
import { translate } from 'react-i18next';

export class GlobalFragmentAuthorization extends Component<Props> {
  render() {
    const {
      account,
      authorization,
      pubkey,
      t
    } = this.props;
    let element = (
      <span>
        {account}
      </span>
    );
    if (authorization) {
      element = (
        <span>
          <span>
            {account}
          </span>
          <span style={{ opacity: 0.4 }}>
            @{authorization}
          </span>
        </span>
      );
    }
    if (pubkey) {
      return (
        <Popup
          content={pubkey}
          header={t('public_key')}
          hoverable
          inverted
          trigger={element}
        />
      );
    }
    return element;
  }
}

export default translate('global')(GlobalFragmentAuthorization);
