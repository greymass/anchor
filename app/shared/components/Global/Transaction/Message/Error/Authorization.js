import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { List } from 'semantic-ui-react';
import ReactJson from 'react-json-view';

class GlobalTransactionMessageErrorAuthorization extends Component<Props> {
  render() {
    const {
      error,
      t
    } = this.props;

    const colorStyle = { color: '#3C3A3B' };

    return (
      <div>
        <p>
          {t('global_transaction_message_error_authorization_paragraph')}
        </p>
        <hr style={colorStyle} />
        <List style={colorStyle} ordered>
          <List.Item>
            {t('global_transaction_message_error_authorization_list_item_one')}
          </List.Item>
          <List.Item>
            {t('global_transaction_message_error_authorization_list_item_two')}
          </List.Item>
          <List.Item>
            {t('global_transaction_message_error_authorization_list_item_three')}
          </List.Item>
        </List>
        <hr style={colorStyle} />
        <p style={colorStyle}>*&nbsp;{t('global_transaction_message_error_authorization_note')}</p>
        {(typeof error === 'object') ? (
          <ReactJson
            collapsed={2}
            displayDataTypes={false}
            displayObjectSize={false}
            iconStyle="square"
            name={null}
            src={error}
            style={{ padding: '1em' }}
          />
        ) : ''}
      </div>
    );
  }
}

export default translate('global')(GlobalTransactionMessageErrorAuthorization);
