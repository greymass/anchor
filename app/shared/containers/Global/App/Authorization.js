// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {
  HashRouter,
  Route,
  Switch,
  withRouter
} from 'react-router-dom';
import { Button, Divider, Form, Grid, Header, Icon, Message, Modal, Segment } from 'semantic-ui-react';
import { find, isObject } from 'lodash';

import DangerLink from '../../../containers/Global/DangerLink';
import GlobalSettingsDfuse from '../../../components/Global/Settings/Dfuse';
import { setSetting, setSettingWithValidation } from '../../../actions/settings';
import { setNodeGeneric } from '../../../actions/blockchains';
import { changeModule } from '../../../../modules/main/actions/navigation';

class AuthorizationContainer extends Component<Props> {
  render() {
    const {
      actions,
      blockchain,
      open,
      settings,
      trigger,
      validate,
    } = this.props;
    let errorMessage = (validate.NODE_ERROR) ? validate.NODE_ERROR.message : false;
    if (validate.NODE_ERROR && validate.NODE_ERROR.response && validate.NODE_ERROR.response.data) {
      if (isObject(validate.NODE_ERROR.response.data)) {
        errorMessage = `${validate.NODE_ERROR.response.data.message} (${validate.NODE_ERROR.response.data.code} - ${validate.NODE_ERROR.response.data.trace_id})`;
      } else {
        errorMessage = validate.NODE_ERROR.response.data;
      }
    }
    // if no blockchain is selected, do not display
    if (!blockchain) return false;
    return (
      <Modal
        centered={false}
        closeIcon
        dimmer="blurring"
        onClose={this.props.onClose}
        onOpen={this.props.onOpen}
        open={open}
        trigger={trigger}
      >
        <Header
          attached="top"
          block
          size="huge"
        >
          <Icon color="orange" name="lock" />
          <Header.Content>
            This API requires a valid API Key
            <Header.Subheader>
              <DangerLink
                content={settings.node}
                link={settings.node}
              />
              {' '}
              is responding but requires authentication.
            </Header.Subheader>
          </Header.Content>
        </Header>
        <Modal.Content>
          <Grid divided>
            <Grid.Row centered>
              <Grid.Column width={12}>
                <Segment>
                  <Form style={{ paddingBottom: '1em' }}>
                    <Header>
                      Enter your API Key
                      <Header.Subheader>
                        In order to use this API server, a valid API key must be defined.
                      </Header.Subheader>
                    </Header>
                    <Form.Field>
                      <label>API Key:</label>
                      <GlobalSettingsDfuse
                        actions={actions}
                        value={settings.dfuseKey}
                      />
                    </Form.Field>
                  </Form>
                  {(errorMessage)
                    ? (
                      <Message
                        content={errorMessage}
                        header="Connection Error"
                        icon="warning circle"
                        error
                        size="large"
                      />
                    )
                    : false
                  }
                  {(validate.NODE === 'PENDING')
                    ? (
                      <Message
                        color="green"
                        header="Connecting"
                        content="Attempting to connect..."
                        icon="loading notched circle"
                        size="large"
                      />
                    )
                    : false
                  }
                  <Button
                    color="green"
                    content="Retry Connection"
                    fluid
                    icon="refresh"
                    loading={validate.NODE === 'PENDING'}
                    onClick={this.props.onRetry}
                  />
                </Segment>

              </Grid.Column>
            </Grid.Row>
            <Grid.Row centered>
              <Grid.Column width={12}>
                <Header>
                  Change API Servers
                </Header>
                <Segment basic>
                  <p>To specify a new blockchain API server to connect with, click the button below and enter a new server in the <strong>Default Node</strong> field.</p>
                  <Button
                    color="blue"
                    content="Edit Blockchain"
                    fluid
                    icon="pencil"
                    onClick={this.props.manageBlockchains}
                  />
                </Segment>
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </Modal.Content>
        <Modal.Actions>
          <Button
            color="orange"
            content="Continue Disconnected"
            onClick={this.props.onClose}
          />
        </Modal.Actions>
      </Modal>
    );
  }
}

function mapStateToProps(state) {
  return {
    blockchain: find(state.blockchains, { chainId: state.settings.chainId }),
    connection: state.connection,
    settings: state.settings,
    validate: state.validate,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      changeModule,
      setNodeGeneric,
      setSetting,
      setSettingWithValidation,
    }, dispatch)
  };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(AuthorizationContainer));
