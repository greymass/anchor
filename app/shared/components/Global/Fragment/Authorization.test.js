import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalFragmentAuthorization } from './Authorization';

const mockProps = {
  account: {},
  authorization: {},
  pubkey: 'EOSXXXX',
  t: jest.fn,
};

describe('GlobalFragmentAuthorization', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalFragmentAuthorization {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
