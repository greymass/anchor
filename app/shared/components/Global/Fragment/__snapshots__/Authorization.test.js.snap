// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFragmentAuthorization renders correctly 1`] = `
<Popup
  content="EOSXXXX"
  header={[MockFunction]}
  hoverable={true}
  inverted={true}
  keepInViewPort={true}
  on="hover"
  position="top left"
  trigger={
    <span>
      <span>
        Object {}
      </span>
      <span
        style={
          Object {
            "opacity": 0.4,
          }
        }
      >
        @
        Object {}
      </span>
    </span>
  }
/>
`;
